### ISSUE TITLE
[Feature Request] Batch interrogote and add captions

### ISSUE THREAD

**Issue by LiJT** [toshiaki1729/stable-diffusion-webui-dataset-tag-editor#51]
It would be nice to add a button that caption image in batch automatically inside dataset tag editor, Hopefully it is easy to implement this feature?

Because currently only way to do so is to do it image by image its very time consuming

Thank you Toshiaki ur plugin works wonderfully. Thanks for your hardwork!

**Comment by toshiaki1729** [comment]
It is the same as pushing batch interrogate images button, to save the current tags and reload dataset with interrogator tags prepended/appended.

**Comment by yggdrasil75** [comment]
when loading, does it actually interrogate in batches, or does it interrogate as a foreach? it doesnt stress any components of my computer, so it doesnt appear to be running in parallel, which is where I think lijt is coming from. if I want to interrogate 5000 images, I start it in the morning and its done a little after I get off work. I could lower the resolution to get it to run faster, but that means that details will not be interrogated and lowers the accuracy of the resultant tags.

**Comment by toshiaki1729** [comment]
> when loading, does it actually interrogate in batches, or does it interrogate as a foreach?

In terms of parallel processing, it does just as a foreach, not using batch inference feature of the models.

If the parallel inference is needed, the interrogator models need to be modified, especially for wd-v14-tagger.

**Comment by yggdrasil75** [comment]
the interrogator model not supporting batching aside, can you run multiple instances of the interrogator at once to effectively batch process? it would be slower than running 1 instance of the interrogator as a single native batch process probably, but it would still be faster than 1 at a time.

**Comment by LiJT** [comment]
> Currently, the images can be interrogated in batch when they are loaded, using some different interrogators at the same time.

Thank you I got it to work!

**Comment by toshiaki1729** [comment]
It seems that original issue is solved, but the new issue related is raised by @yggdrasil75.
I opened #55 and will close this issue.
